### PR TITLE
Handle external 'prop-types' correctly in prod build

### PR DESCRIPTION
**Pull request by Marc-Andre-Rivet**
This PR correctly externalizes `prop-types` for usage as a library. It also exposes the library as a prop on `window` to be consistent with other Dash libraries.

Would normally not update the build as part of a PR but this is included in another project through package.json and makes everything easier to handle on the other end.

`"dash-table": "github:plotly/dash-table#prod-prop-types"`

### CODE REVIEW

File: .config/webpack/base.js [hunk 1]
```diff
@@ -19,29 +19,19 @@ module.exports = (options = {}) => {
...
-                root: 'React'
-            },
-            'react-dom': {
-                commonjs: 'react-dom',
-                commonjs2: 'react-dom',
-                amd: 'ReactDOM',
-                root: 'ReactDOM'
-            },
+            react: 'React',
+            'react-dom': 'ReactDOM',
```
Comment (Marc-Andre-Rivet) [inline .config/webpack/base.js]: This can be simplified as it's target is `window` now

File: .config/webpack/base.js [hunk 2]
```diff
@@ -19,29 +19,19 @@ module.exports = (options = {}) => {
...
-            demo: ['./demo/index.js', './demo/index.html']
+            demo: ['./demo/index.html', './demo/index.js']
         },
         mode: mode,
         output: {
             path: path.resolve(__dirname, `./../../${dashLibraryName}`),
             filename: '[name].js',
             library: dashLibraryName,
-            libraryTarget: 'umd'
+            libraryTarget: 'window'
```
Comment (Marc-Andre-Rivet) [inline .config/webpack/base.js]: Loaded unto `window.dash_table` in the browser

Comment (wbrgss) [reply .config/webpack/base.js]: Will changing how this is exposed have any side-effects for non-Dash users? Could you imagine anyone using RequireJS or anything else that expects a `umd` in their JavaScript project? I suppose, obviously, we wouldn't expect Node.js users (without a `window` global namespace) to be using this project. I'm also not sure if there's any webpack magic that makes this a non-issue.

Comment (Marc-Andre-Rivet) [reply .config/webpack/base.js]: This should not have an impact.

**Comment by Marc-Andre-Rivet** [comment]
@wbrgss Yes. It will fix the missing prop types dep

**Comment by Marc-Andre-Rivet** [comment]
Adding a changelog entry and will merge afterwards.

**Pull request merged**